PEP 667: Add discussion and implementation link (#3688)
python/peps: peps/pep-0667.rst

Title: Consistent views of namespaces
Author: Mark Shannon <mark@hotpy.org>,
        Tian Gao <gaogaotiantian@hotmail.com>
Discussions-To: https://discuss.python.org/t/46631
Status: Draft
Type: Standards Track

> Content-Type: text/x-rst

Reviewer: We don't need these any more, now PEPs all have .rst extensions:
```suggestion
```

This is inconsistent, and confusing.
With this PEP both examples would print ``2``.

Worse than that, the current behavior can result in strange `bugs

> <https://bugs.python.org/issue30744>`__.

Reviewer: Link to its new home:
```suggestion
<https://github.com/python/cpython/issues/74929>`__.
```

``PyEval_GetLocals()`` already creates a cycle, as it returns a borrowed reference.

Implementation
==============

> The implementation is in development as a draft pull request on GitHub ([2]_).

Reviewer: We prefer inline links where there's no additional info listed in the references, it saves people a click.

```suggestion
The implementation is in development as a `draft pull request on GitHub
<https://github.com/python/cpython/pull/115153>`__.
```

> References
> ==========
> 
> .. [1] https://bugs.python.org/issue30744
> 
> .. [2] https://github.com/python/cpython/pull/115153
> 

Reviewer: Please could you also turn [1] into an inline link as well?

And link it to `https://github.com/python/cpython/issues/74929` instead of the old bugs.python.org.

```suggestion
```
